src/modules/Api.ts

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2025 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only OR LicenseRef-Element-Commercial
+Please see LICENSE files in the repository root for full details.
+*/
+
+import type { Api, RuntimeModuleConstructor, Config } from "@element-hq/element-web-module-api";
+import { ModuleRunner } from "./ModuleRunner.ts";
+import AliasCustomisations from "../customisations/Alias.ts";
+import { RoomListCustomisations } from "../customisations/RoomList.ts";
+import ChatExportCustomisations from "../customisations/ChatExport.ts";
+import { ComponentVisibilityCustomisations } from "../customisations/ComponentVisibility.ts";
+import DirectoryCustomisations from "../customisations/Directory.ts";
+import LifecycleCustomisations from "../customisations/Lifecycle.ts";
+import * as MediaCustomisations from "../customisations/Media.ts";
+import UserIdentifierCustomisations from "../customisations/UserIdentifier.ts";
+import { WidgetPermissionCustomisations } from "../customisations/WidgetPermissions.ts";
+import { WidgetVariableCustomisations } from "../customisations/WidgetVariables.ts";
+import SdkConfig from "../SdkConfig.ts";
+
+const legacyCustomisationsFactory = <T extends object>(baseCustomisations: T) => {
+    let used = false;
+    return (customisations: T) => {
+        if (used) throw new Error("Legacy customisations can only be registered by one module");
+        Object.assign(baseCustomisations, customisations);
+        used = true;
+    };
+};
+
+class ConfigApi {
+    public get(): Config;
+    public get<K extends keyof Config>(key: K): Config[K];
+    public get<K extends keyof Config = never>(key?: K): Config | Config[K] {
+        if (key === undefined) {
+            return SdkConfig.get() as Config;
+        }
+        return SdkConfig.get(key);
+    }
+}
+
+class ModuleApi implements Api {
```

Comment:
This could do with some doc probably, since it is being exported below (also I'm not entirely sure what the API of this thing is)

Reply:
The more detailed docs are in `@element-hq/element-web-module-api` - given that is reused by both sides it makes more sense to be the home of the docs